### ISSUE TITLE
Drop support for Python 3.4

### ISSUE THREAD

**Issue by jhamman** [pydata/xarray#1829]
Python 3.7-final is due out in June ([PEP 537](https://www.python.org/dev/peps/pep-0537/)). When do we want to deprecate 3.4 and when should we drop support all together. @maxim-lian brought this up in a PR he's working on: https://github.com/pydata/xarray/pull/1828#issuecomment-357562144.

For reference, we dropped Python 3.3 in #1175 (12/20/2016).

**Comment by shoyer** [comment]
It looks like pandas has already dropped Python 3.4 support in v0.21: https://github.com/pandas-dev/pandas/issues/18763

So I'd be favor of dropping it for xarray too in the next major release.

**Comment by Zac-HD** [comment]
Related: when does Xarray plan to drop Python 2?  IMO we should *at least* join [python3statement.org](http://www.python3statement.org) and drop it by 2020, and clearly document the timeline whatever it happens to be.

I'd be keen for an earlier date TBH - keyword-only arguments are great, compatibility shims kinda suck, and dependencies are moving to py3-only at an increasing rate (including matplotlib 3.0, scheduled for July) - but can see the other argument too.

**Comment by jhamman** [comment]
Saving the PR until after 0.10 is released but here's commit that drops Python 3.4: 5c96b54929622d549ac059499bec4c32ddfee1ce

**Comment by maxim-lian** [comment]
@jhamman shall we do this now? Amazingly this is breaking the flake8 of https://github.com/pydata/xarray/pull/2031

**Comment by jhamman** [comment]
Fine by me. Would we want to make the next version 0.11 then?

**Comment by shoyer** [comment]
If maintaining Python 3.4 support is a pain, I'm happy to drop it (certainly from our test suite). I would rather save major versions for major revisions to project, however.

**Comment by maxim-lian** [comment]
Completely agree re versions, and I can navigate around it; assuming 0.11 is not the next release

**Comment by maxim-lian** [comment]
Completely agree re versions, and I can navigate around it; assuming 0.11 is not the next release

**Comment by shoyer** [comment]
Indeed, my plan is to release 0.10.3 shortly after your `isin` PR gets in :)

**Comment by jhamman** [comment]
Apologies for conflating the two issues here. I didn't mean to suggest we should skip straight to 0.11 for the sake of dropping 3.4. More that we should try to time the two together if possible. As @shoyer said, we can drop 3.4 from the test suite if need be.

**Comment by davidh-ssec** [comment]
I just ran in to an issue testing Python 3.4 on Travis where xarray asked for pandas >0.18.0 which pulls in a version of pandas that is not compatible with Python 3.4 (https://github.com/pandas-dev/pandas/issues/20697). It also seems like this could be related to pip 10.0.

I'm ok dropping Python 3.4 from my tests, but is this python version check something pip/pypi should handle or is it something that xarray has to check in its setup.py?

**Comment by shoyer** [comment]
Python 3.4 was supported by pandas 0.18, so I'm not sure what's going on here. I don't think it's really an xarray issue.

**Comment by davidh-ssec** [comment]
Looks like it is related to pip 10.0, with pip 9.0.3 it seems to install pandas fine on Python 3.4. I'll continue debugging this with pandas and the pip projects. Thanks.
